### DISCUSSION TITLE
How to create parameter "metapath" in MetaPath2Vec model on the dataset with same dst and src node type

### DISCUSSION THREAD

**Discussion by qinguangming1999** [pyg-team/pytorch_geometric#8660] [Q&A] [answered]
For Aminer the metapath is 
metapath = [
    ('author', 'writes', 'paper'),
    ('paper', 'published_in', 'venue'),
    ('venue', 'publishes', 'paper'),
    ('paper', 'written_by', 'author'),
]
So, what about the dataset lastFM. we have the "User - User" relationship. and it's for the link prediction(user-artist) task.
metapath_lastFM = [
    ('user', 'relation0', 'user'),
    ('user', 'relation0', 'user'),
    ('user', 'relation1', 'artist'),
    ('artist', 'relation1', 'user'),
].
is the metapath_lastFM correct?

**Comment by rusty1s** [comment, marked as answer]
Yes, that looks good.

**Comment by Jieerbobo** [comment]
metapath_lastFM = [
('user', 'relation0', 'user'),
('user', 'relation0', 'user'),
('user', 'relation1', 'artist'),
('artist', 'relation1', 'user'),
]

Hi, I would like to ask, for the first two edge types, should their edge indexes be the same or taken inverse?
Does this matapath refer to "uuu and uau" or "uuuau"?

**Reply by rusty1s** [reply]
It would be uuuau. Whether you want to use the same edge type twice depends on your graph, e.g., `relation0` might be an undirected edge type.
